cmd/ui.go

```diff
@@ -186,34 +264,55 @@ func showProgress(ctx context.Context, conf Config, execScheduler *local.Executi
 	if stdoutTTY && !noColor {
 		updateFreq = 100 * time.Millisecond
 		outMutex.Lock()
-		stdout.PersistentText = progressBarsPrint
-		stderr.PersistentText = progressBarsPrint
+		stdout.PersistentText = printProgressBars
+		stderr.PersistentText = printProgressBars
 		outMutex.Unlock()
 		defer func() {
 			outMutex.Lock()
 			stdout.PersistentText = nil
 			stderr.PersistentText = nil
-			if ctx.Err() != nil {
-				// Render a last plain-text progressbar in an error
-				progressBarsLastRender = []byte(renderMultipleBars(stdoutTTY, false, leftMax, pbs))
-				progressBarsPrint()
-			}
 			outMutex.Unlock()
 		}()
 	}
 
+	var (
+		fd              = int(os.Stdout.Fd())
+		ticker          = time.NewTicker(updateFreq)
+		updateTermWidth func()
+		winchSignal     = getWinchSignal()
+	)
+	// On platforms that have SIGWINCH (*nix), progress bar resizing will be
+	// more responsive. On others (Windows), we fallback to resize on every tick.
+	if winchSignal != nil {
+		winch := make(chan os.Signal, 1)
+		signal.Notify(winch, winchSignal)
+		updateTermWidth = func() {
+			select {
+			case <-ticker.C:
```

Comment:
you could just as well only initialize this in the `else` clause ;)

Reply:
Hhmm not sure if I understood you correctly, but let me know if 17c92555 resolves this.